sklearn/preprocessing/_encoders.py

```diff
@@ -41,6 +41,9 @@ def _check_X(self, X):
           not do that)
 
         """
+        if hasattr(X, 'iloc'):
+            # if pandas dataframes
```

Comment:
Can you add some comment here about why we skip the check for pandas? (because it will be done column by column later on). 
Or maybe add it to the docstring above.

Reply:
I added a comment (as above)
and check if instead of pandas DataFrame Sequence is not passed 
and a couple of tests